### PR TITLE
[Dagger] Use @IsPocketInstalled dagger qualifier

### PR DESCRIPTION
**Pull request by pt2121**
re #537 

* adding `IsPocketInstalled` qualifier
* ~adding `FeedColumns` qualifier~
* ~injecting `FeedAdapter`~

### CODE REVIEW

File: app/src/main/java/io/plaidapp/ui/HomeActivity.java [hunk 1]
```diff
@@ -113,7 +113,9 @@
     private int columns;
     private TextView noFiltersEmptyText;
     private FilterAdapter filtersAdapter;
-    private FeedAdapter adapter;
+
+    @Inject
```
Comment (florina-muntenescu) [inline app/src/main/java/io/plaidapp/ui/HomeActivity.java]: We don't need to inject the FeedAdapter, it's ok to let the HomeActivity create it.

File: core/src/main/java/io/plaidapp/core/dagger/qualifier/FeedColumns.kt [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ */
+
+package io.plaidapp.core.dagger.qualifier
+
+import javax.inject.Qualifier
+import kotlin.annotation.AnnotationRetention.BINARY
+
+@Retention(BINARY)
+@Qualifier
+annotation class FeedColumns
```
Comment (florina-muntenescu) [inline core/src/main/java/io/plaidapp/core/dagger/qualifier/FeedColumns.kt]: new line

File: core/src/main/java/io/plaidapp/core/dagger/qualifier/IsPocketInstalled.kt [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+ */
+
+package io.plaidapp.core.dagger.qualifier
+
+import javax.inject.Qualifier
+import kotlin.annotation.AnnotationRetention.BINARY
+
+@Retention(BINARY)
+@Qualifier
+annotation class IsPocketInstalled
```
Comment (florina-muntenescu) [inline core/src/main/java/io/plaidapp/core/dagger/qualifier/IsPocketInstalled.kt]: new line

File: core/src/main/java/io/plaidapp/core/feed/FeedAdapter.java [hunk 4]
```diff
@@ -101,9 +104,10 @@
     private List<PlaidItem> items;
     private boolean showLoadingMore = false;
 
+    @Inject
```
Comment (florina-muntenescu) [inline core/src/main/java/io/plaidapp/core/feed/FeedAdapter.java]: don't inject the FeedAdapter.

File: search/src/main/java/io/plaidapp/search/ui/SearchActivity.kt [hunk 5]
```diff
@@ -90,7 +89,8 @@ class SearchActivity : AppCompatActivity() {
     private val transitions = SparseArray<Transition>()
     private var focusQuery = true
 
-    private lateinit var feedAdapter: FeedAdapter
+    @Inject
```
Comment (florina-muntenescu) [inline search/src/main/java/io/plaidapp/search/ui/SearchActivity.kt]: As in home, don't inject the FeedAdapter, let the SearchActivity create it.

**Comment by manuelvicnt** [comment]
I personally prefer the `@Named` annotation to solve this kind of problem. The naming can be more concise and less code to write. I'm using it in #649 in case you want to take a look.

Should we move away from Qualifiers and use `@Named`?

**Comment by florina-muntenescu** [comment]
> I personally prefer the `@Named` annotation to solve this kind of problem. The naming can be more concise and less code to write. I'm using it in #649 in case you want to take a look.
> 
> Should we move away from Qualifiers and use `@Named`?

@pt2121 - let's use this approach. Thanks!

**Comment by ZacSweers** [comment]
Commented on the linked PR as well, but Named should be the exception and not the standard IMO.

Custom qualifiers can:
- be documented
- contain extra fields
- be stripped by proguard via non-runtime retention
- don't require sharing and reusing a string constant for consistency (which Named conventionally should use to avoid errors)
- can have custom visibility (package/protected/private) to limit access

And personal preference - I find `@ForPocket isPocketInstalled: Boolean` nicer/simpler to read than `@Named("ForPocket") isPocketInstalled: Boolean` or `@Named(FOR_POCKET) isPocketInstalled: Boolean`. The qualifier is just one element - the name. With Named, it's an element and its value that you have to from at a glance.

File: app/src/main/java/io/plaidapp/dagger/HomeModule.kt [hunk 6]
```diff
@@ -61,10 +63,12 @@ abstract class HomeModule {
     @Module
     companion object {
 
+        @FeedColumns
```
Comment (florina-muntenescu) [inline app/src/main/java/io/plaidapp/dagger/HomeModule.kt]: Let's stick with the initial scope of #537 and just use the qualifier for `IsPocketInstalled`. The columns won't be retrieved anywhere else than in the activity and we're not injecting them in the adapter, so I don't think there's a need to inject it.

**Comment by manuelvicnt** [comment]
@ZacSweers made really good points and also, @ronshapiro in the other PR (#649). Seems we should go for that approach. So it's ok if you leave the implementation as it is :) I'll change mine! Thank you

**Comment by pt2121** [comment]
thanks y'all for the feedback. updated my PR based on the comments

**Comment by pt2121** [comment]
@florina-muntenescu , could you take another look when having a chance?
I think I addressed all comments. thx!

**Comment by florina-muntenescu** [comment]
Looks good to me. 
@ZacSweers would you mind taking a look as well, please?

File: search/src/main/java/io/plaidapp/search/ui/SearchActivity.kt [hunk 7]
```diff
@@ -95,6 +95,10 @@ class SearchActivity : AppCompatActivity() {
     @Inject
     lateinit var viewModel: SearchViewModel
 
+    @JvmField
+    @field:[Inject IsPocketInstalled]
```
Comment (ZacSweers) [inline search/src/main/java/io/plaidapp/search/ui/SearchActivity.kt]: I always forget you can do this notation, neat!

**Comment by florina-muntenescu** [comment]
Thanks for the PR @pt2121 !

**Pull request merged**